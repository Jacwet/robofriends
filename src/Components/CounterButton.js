import React, { Component } from "react";

class CounterButton extends Component{
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
      color: 'blue'
    }
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true
    }
    return false;
}
  render() {
    console.log('CounterButton');
    return (
      <button
        id="counter"
        color={this.props.color}
        onClick={() => this.setState(state => ({count: state.count + 1}))}
      >
        <h1 >Count: {this.state.count}</h1>
        <h1 >Color: {this.state.color}</h1>
      </button>
  )
}

}

export default CounterButton;