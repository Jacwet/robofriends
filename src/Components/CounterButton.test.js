import React from 'react';
import {render, fireEvent, screen, wrapper, getByText } from '@testing-library/react';
import '@testing-library/jest-dom';
import CounterButton from './CounterButton';


const mockColor = 'red'

it('renders', () => {
  const { asFragment } = render(<CounterButton color={mockColor} />);
  expect(asFragment()).toMatchSnapshot();
});

it('correctly incraments the counter', async () => {
  render(<CounterButton color={mockColor} />)
  const button = screen.getByRole('button')
  await button.click();
  expect(button).toHaveTextContent("1")

})

it('correctly incraments the counter2', async () => {
  render(<CounterButton color={'red'} />)
  const button = screen.getByRole('button')
  expect(button).toHaveTextContent("blue")
})