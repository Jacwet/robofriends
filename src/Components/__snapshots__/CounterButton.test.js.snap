// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    color="red"
    id="counter"
  >
    <h1>
      Count: 0
    </h1>
    <h1>
      Color: blue
    </h1>
  </button>
</DocumentFragment>
`;
